### PR TITLE
Updating CQUI files for Spring 2018 Update - issue #331

### PR DESCRIPTION
**Pull request by fvdmagalhaes**
Updating CQUI files for Spring 2018 update (1.0.0.257).

The following files were updated:
- Assets/UI/Panels/citypaneloverview.lua
- Assets/UI/Panels/citypaneloverview.xml
- Assets/UI/Panels/notificationpanel.lua
- Assets/UI/Popups/techciviccompletedpopup.lua
- Assets/UI/Screens/techtree.lua
- Assets/UI/WorldView/citybannermanager.xml
- Assets/UI/actionpanel.lua
- Assets/UI/partialscreenhooks.lua
- Assets/UI/unitflagmanager.lua

### CODE REVIEW

File: Assets/UI/actionpanel.lua [hunk 1]
```diff
@@ -565,7 +565,8 @@ function DoEndTurn( optionalNewBlocker:number )
 
   -- If the player can unready their turn, request that.
   -- CanUnreadyTurn() is only true when the player should unready their turn instead of turn completing.
-  if pPlayer:CanUnreadyTurn() then
+  if pPlayer:CanUnreadyTurn()
+	and not UI.IsTurnTimerElapsed(Game.GetLocalPlayer()) then
```
[the remark this answers is not in the document]
Comment (fvdmagalhaes) [reply Assets/UI/actionpanel.lua]: @ihendriks , thanks for pointing this. I'm going to adjust the indentation.

File: Assets/UI/partialscreenhooks.lua [hunk 2]
```diff
@@ -241,14 +254,12 @@ end
...
-  --  Note: I'm using not IsMajor and not IsBarbarian to determine the city-state check at this point - though elsewhere we have used CanReceiveInfluencePoints.
-  --  For future expansions we should probably create an IsMinor check for the Player object.
   if (not m_isCityStatesUnlocked) then
     local localDiplomacy:table = localPlayer:GetDiplomacy();
     local aPlayers = PlayerManager.GetAliveMinors();
     for _, pPlayer in ipairs(aPlayers) do
-      if (not pPlayer:IsMajor() and not pPlayer:IsBarbarian() and localDiplomacy:HasMet(pPlayer:GetID())) then
-        m_isCityStatesUnlocked = true;
+	  if (pPlayer:IsMinor() and localDiplomacy:HasMet(pPlayer:GetID())) then
+		m_isCityStatesUnlocked = true;
```
Comment (ihendriks) [inline Assets/UI/partialscreenhooks.lua]: The last one. I think I missed this last time as well :(

**Comment by fvdmagalhaes** [comment]
Updating two more files: notificationpanel.lua and citybannermanager.xml

File: Assets/UI/Panels/citypaneloverview.lua [hunk 3]
```diff
@@ -281,6 +281,9 @@ function ViewPanelBreakdown( data:table )
   for _, wonder in ipairs(data.Wonders) do
     local kInstanceWonder:table = m_kWondersIM:GetInstance();
     kInstanceWonder.WonderName:SetText( wonder.Name );
+  local pRow = GameInfo.Buildings[wonder.Type];
+  local sToolTip = ToolTipHelper.GetBuildingToolTip( pRow.Hash, playerID, m_pCity );
+  kInstanceWonder.Top:SetToolTipString( sToolTip );
```
Comment (Azurency) [inline Assets/UI/Panels/citypaneloverview.lua]: Indentation not right here

File: Assets/UI/Panels/citypaneloverview.xml [hunk 4]
```diff
@@ -40,6 +40,10 @@
...
                   <Label  ID="FoodConsumption"            Anchor="R,C"                                      Style="CityPanelTextSmall"      String="-" />
                 </Grid>
+                <Grid                                                  Offset="0,40"                        Style="CityPanelLineItem">
+                  <Label                                  Anchor="L,C"                                      Style="CityPanelTextSmall"    String="LOC_HUD_CITY_FOOD_NEEDED_FOR_GROWTH" />
+                  <Label  ID="FoodNeededForGrowth"        Anchor="R,C"                                      Style="CityPanelTextSmall"    String="-" />
+                </Grid>
               </Grid>
               <Grid                                                    Offset="0,0"     Size="parent,8"     Style="Divider3Grid"  />
               <Grid                                       Anchor="L,T" Offset="0,-2"    Size="parent,auto" Style="CityPanelSubPanelContainer" InnerPadding="0,10" InnerOffset="0,5">
```
Comment (Azurency) [inline Assets/UI/Panels/citypaneloverview.xml]: missing one change line 84 in label `id="GrowthLongNum"`, offset is `4,-2` now

File: Assets/UI/actionpanel.lua [hunk 5]
```diff
@@ -565,9 +565,10 @@ function DoEndTurn( optionalNewBlocker:number )
 
   -- If the player can unready their turn, request that.
   -- CanUnreadyTurn() is only true when the player should unready their turn instead of turn completing.
-  if pPlayer:CanUnreadyTurn() then
-    UI.RequestAction(ActionTypes.ACTION_UNREADYTURN);
-    return;
+  if pPlayer:CanUnreadyTurn()
+    and not UI.IsTurnTimerElapsed(Game.GetLocalPlayer()) then
+      UI.RequestAction(ActionTypes.ACTION_UNREADYTURN);
```
Comment (Azurency) [inline Assets/UI/actionpanel.lua]: this is over indented, should only be 4 spaces, same thing for the next line.

File: Assets/UI/Panels/notificationpanel.lua [hunk 6]
```diff
@@ -712,63 +716,66 @@ function OnDefaultAddNotification( pNotification:table )
...
-      if (notificationEntry.m_IconName ~= nil) then
-        local textureOffsetX, textureOffsetY, textureSheet = IconManager:FindIconAtlas(notificationEntry.m_IconName,40);
-        if (textureOffsetX ~= nil) then
-            notificationEntry.m_Instance.Icon:SetTexture( textureOffsetX, textureOffsetY, textureSheet );
+  -- The icon is displayable in the current mode.
+  if ( table.count(notificationEntry.m_IDs)==1 
+    and pNotification:GetEndTurnBlocking() == EndTurnBlockingTypes.NO_ENDTURN_BLOCKING
+    and pNotification:IsIconDisplayable() ) then
+
+      notificationEntry.m_Instance    = m_genericItemIM:GetInstance();
```
Comment (Azurency) [inline Assets/UI/Panels/notificationpanel.lua]: This diff is kind of messy, why did you over-indent this whole block ? It should be 4 spaces like it was before, no need to change that, just update the `if` statement 👍

File: Assets/UI/Popups/techciviccompletedpopup.lua [hunk 7]
```diff
@@ -260,30 +259,18 @@ function ShowNextQueuedPopup()
...
-    else
-      -- Add to queue if already showing a tech/civic completed popup
-      table.insert(m_kQueuedPopups, techCompletedEntry);
-    end
+      if not m_isWaitingToShowPopup then
+        ShowCompletedPopup(completedEntry);    
+      else
+        -- Add to queue if already showing a tech/civic completed popup
+        table.insert(m_kQueuedPopups, completedEntry);
+      end
```
Comment (Azurency) [inline Assets/UI/Popups/techciviccompletedpopup.lua]: This whole block is also over-indented, should be 4 spaces.

**Comment by fvdmagalhaes** [comment]
The files were updated according to @Azurency observations. Thanks.

**Comment by Azurency** [comment]
Thanks, it seems all good now ! 
And thank you for your first contribution 🎉 it's really appreciated !

**Pull request merged**